File: src/part01/chapter02/levelC/matrix/java/MatrixCreator.java
package part01.chapter02.levelC.matrix.java;

import java.util.Random;
import java.util.Scanner;

public class MatrixCreator {

    public double[][] getSquareMatrix() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Please input N-value for matrix N*N, N should be equals or more 1");
        final int N = scanner.nextInt();
        if (N < 1) {
            System.out.println("Incorrect N: N should be equals or more 1");
            return getSquareMatrix();
        }

        double[][] result = new double[N][N];
        for (int indexRow = 0; indexRow < N; indexRow++) {
            for (int indexElementInRow = 0; indexElementInRow < N; indexElementInRow++) {
                result[indexRow][indexElementInRow] = getRandomDoubleInRange(-N, N);
            }
        }

        printSquareMatrix(result);
        return result;
    }

    private void printSquareMatrix(double[][] squareMatrix) {
        System.out.println("Matrix generated:");
        MatrixPrinter.printMatrix(squareMatrix);
    }

    private double getRandomDoubleInRange(double min, double max) {
        return min + (max - min) * new Random().nextDouble();
    }
}

File: src/part01/chapter02/levelC/matrix/kotlin/MatrixCreator.kt
package part01.chapter02.levelC.matrix.kotlin

import part01.chapter02.levelC.console.kotlin.readInt
import kotlin.random.Random

object MatrixCreator {

    fun getSquareMatrix(): Array<DoubleArray> {
        println("Please input N-value for matrix N*N, N should be equals or more 1")
        val n = readInt()
        if (n <= 1) {
            println("Incorrect N: N should be equals or more 1")
            return getSquareMatrix()
        }

        val result = Array(n) { DoubleArray(n) }
        for (rowIndex in result.indices) {
            for (columnIndex in result[rowIndex].indices) {
                result[rowIndex][columnIndex] = Random.nextDouble(-n.toDouble(), n.toDouble())
            }
        }

        printGeneratedSquareMatrix(result)
        return result
    }

    private fun printGeneratedSquareMatrix(squareMatrix: Array<DoubleArray>) {
        println("Matrix generated:")
        printMatrix(squareMatrix)
    }

}